src/Files.App/Helpers/UIHelpers.cs

```diff
@@ -143,5 +144,14 @@ private static IEnumerable<IconFileInfo> LoadSidebarIconResources()
 
             return imageResList;
         }
+
+        public static int GetAdaptedRecycleBinIconIndex()
+        {
+            if (new RecycleBinHelpers().RecycleBinHasItems())
```

Comment:
Can you avoid creating new instances?

Reply:
Currently no, the while RecycleBinHelper is kind of a mess to that regard. There is static and not static methods. But it's a good time as any to make a refactoring of that helper.

Reply:
Made RecycleBin a static class, think it will address the issue.